core/spec/models/spree/variant_spec.rb

```diff
@@ -811,12 +811,56 @@
     end
   end
 
-  describe '#gallery' do
-    let(:product) { Spree::Variant.new }
-    subject { product.gallery }
+  describe "#gallery" do
+    let(:variant) { build_stubbed(:variant) }
+    subject { variant.gallery }
 
-    it 'responds to #images' do
+    it "responds to #images" do
       expect(subject).to respond_to(:images)
     end
+
+    context "when variant.images is empty" do
+      let(:product) { create(:product) }
+      let(:variant) { create(:variant, product: product) }
+
+      it "fallbacks to variant.product.master.images" do
+        product.option_types = [create(:option_type)]
+        product.master.images = [create(:image)]
+        product.taxons = [create(:taxon)]
+        product.properties = [create(:property)]
```

Comment:
Why do we need to set all these associations here?

Reply:
We didn't, copy+paste mistake... Fixed it :smile: